chapter/related-work.md

```diff
@@ -6,6 +6,93 @@ As such, it is important to get a picture of the field of current LSP projects.
+LLVM performs optimization of the IR and passes it to a compiler backend which in turn generates bytecode for specific architectures e.g. `x86_64`, `MIPS`, `aarch64`, `wasm`, etc.
+Notably through this mechanism, languages gain support for a range of architectures and profit from existing optimizations developed for the IR.
+
+With Monto, Kreidel et al propose a similar idea for IDE portability.
+The paper describes the *Monto IR* and how they use a *Message Broker* to receive events from the Editor and dispatch them to *Monto Services*.
+
+The Monto IR is a language-agnostic and editor-independent tree-like model serialized as JSON.
+Additionally, the IR maintains low level syntax highlighting information (font, color, style, etc.) but leaves the highlighting to the language specific service.
+
+The processing and modification of the source code and IR is performed by *Monto Services*.
+Services implement specific actions, e.g. parsing, outlining or highlighting.
+A central broker connects the services with each other and the editor.
+
+Since Monto performs all work on the IR, independent of the editor, and serializes the IR as JSON messages, the language used to implement *Monto Services* can be chosen freely giving even more flexibility.
+
+The Editor extension's responsibility is to act as a source and sink for data.
+It sends Monto compliant messages to the broker and receives processing results such as (error) reports.
+The communication is based on the ZeroMQ[zeromq] technology which was chosen because it is lightweight and available in manly languages [@monto-disintegrated] allowing to make use of existing language tools.
+
+#### Merlin
+
+The Merlin tool [@merlin,@merlin-website] is in many ways a more specific version of the idea presented in Monto.
+Merlin is a language server for the Ocaml language, yet predates the Language Server Protocol.
+
+The authors of Merlin postulate that implementing "tooling support traditionally provided by IDEs" for "niche languages" demands to "share the language-awareness logic" between implementations.
+As an answer to that, they describe the architecture of Merlin in [@merlin].
+
+Similarly to Monto, Merlin separates editor extensions from language analysis.
+However, Merlin uses a command line interface instead of message passing for interaction.
+Editor extensions expose the server functions to the user by integrating with the editor.
+
+The Merlin server hand provides a single optimized implementation of code intelligence for Ocaml.
+Since all resources could be put to a single project, multiple iterations of performance improvements were done on Merlin.
+It now supports partial, incremental parsing and type-checking which allows users to query information even about incomplete or incorrect programs.
```

Comment:
```suggestion
Thanks to this architecture, the Merlin developers have been able to focus their efforts in a single project providing intelligence for OCaml source code.
The result of this work is feature is a platform independent, performant and fault-tolerant provider of language intelligence.
Special care has been taken with the implementation with regards to incremental parsing typechecking and analysis, which allows users to query information even about incomplete or incorrect programs.
```

Reply:
This looks ok, but you may want to reread it, there are several typos (like "is feature is a platform independent").